### PR TITLE
Migrate from attributes to inputs and develop additional controls

### PR DESCRIPTION
**Pull request by kkola**

### CODE REVIEW

**Review by aaronlippold — COMMENTED** [review]
The inspec.yml it's not an attribute file

File: controls/V-75605.rb [hunk 1]
```diff
@@ -51,5 +51,19 @@
...
+
+  if library_files.count > 0
+    library_files.each do |lib_file|
+      describe file(lib_file) do
+        its('group') { should cmp 'root' }
+      end
+    end
+  else
+    describe "No Files Found" do
+      skip "No system-wide shared library files found in /lib, /lib64, or /usr/lib, that are less permissive than 0755"
```
Comment (aaronlippold) [inline controls/V-75605.rb]: This should not skip it should pass the test if they should pass. If the library_files.count == 0 then we have ensured there are not files in those directories that are more permissive than 0755 yes?

This should be updated to give credit for passing the control.

File: controls/V-75607.rb [hunk 2]
```diff
@@ -46,5 +46,19 @@
...
+
+  if library_files.count > 0
+    library_files.each do |lib_file|
+      describe file(lib_file) do
+        its('owner') { should cmp 'root' }
+      end
+    end
+  else
+    describe "No Files Found" do
+      skip "No system-wide shared library files found found in /lib, /lib64, or /usr/lib, that are NOT owned by root"
```
Comment (aaronlippold) [inline controls/V-75607.rb]: Same as above we are skipping where we should be passing  ...

File: controls/V-75609.rb [hunk 3]
```diff
@@ -46,5 +46,19 @@
...
+
+  if library_files.count > 0
+    library_files.each do |lib_file|
+      describe file(lib_file) do
+        its('group') { should cmp 'root' }
+      end
+    end
+  else
+    describe "No Files Found" do
+      skip "No system-wide shared library files found found in /lib, /lib64, or /usr/lib, that are NOT group-owned by root"
```
Comment (aaronlippold) [inline controls/V-75609.rb]: Same as above

File: controls/V-75575.rb [hunk 4]
```diff
@@ -50,5 +50,23 @@
...
+
+  if other_mount_points.count > 0
+    other_mount_points.each do |mount_point|
+      describe mount(mount_point) do
+        its('options') { should include 'nosuid' }
+      end
+    end
+  else
+    describe "No other mount points found" do
+      skip "Separate file system has not been detected for the user home directories"
```
Comment (aaronlippold) [inline controls/V-75575.rb]: So what is the subject in this case, If none are found does it pass or fail?

File: controls/V-75579.rb [hunk 5]
```diff
@@ -40,5 +40,18 @@
...
+  device_rules = etc_fstab.where{ file_system_type == 'nfs' }.entries
+  if device_rules.count > 0
+    device_rules.each do |device_rule|
+      describe device_rule do
+        its ('mount_options') { should include 'nosuid' }
+      end
+    end
+  else
+    describe "No NFS mounts found" do
+      skip "No NFS mounts found on the system"
```
Comment (aaronlippold) [inline controls/V-75579.rb]: Same as before, does it pass or fail on none found?

File: controls/V-75581.rb [hunk 6]
```diff
@@ -41,5 +41,18 @@
...
+  device_rules = etc_fstab.where{ file_system_type == 'nfs' }.entries
+  if device_rules.count > 0
+    device_rules.each do |device_rule|
+      describe device_rule do
+        its ('mount_options') { should include 'noexec' }
+      end
+    end
+  else
+    describe "No NFS mounts found" do
+      skip "No NFS mounts found on the system"
```
Comment (aaronlippold) [inline controls/V-75581.rb]: Same as above

File: controls/V-75583.rb [hunk 7]
```diff
@@ -45,5 +45,20 @@
...
+  directories = command("sudo find / -perm -2 -type d ! -group sys ! -group root ! -group bin -exec ls -lLd {} \\;").stdout.strip.split("\n").entries
+  if directories.count > 0
+    directories.each do |entry|
+      describe directory(entry) do
+        its('group') { should be_in ['root','sys', 'bin'] + application_groups}
+      end
+    end
+  else
+    describe "No world-writable directories found" do
+      skip "No world-writable directories found on the system"
```
Comment (aaronlippold) [inline controls/V-75583.rb]: Give credit for passing

File: controls/V-75585.rb [hunk 8]
```diff
@@ -42,5 +42,19 @@
...
+is_kdump_required = input('is_kdump_required')
+  if is_kdump_required
+    describe service('kdump') do
+      it { should_not be_enabled }
+      it { should_not be_installed }
+      it { should_not be_running }
+    end
+  else
+    describe "kdump service is required and documented with ISSO" do
+      skip ("kdump service is required and documented with ISSO")
```
Comment (aaronlippold) [inline controls/V-75585.rb]: If Kdump is required then it should be running, else KDUMP is not required and this is impact 0 aka Not Applicable or is it a failure?

File: controls/V-75611.rb [hunk 9]
```diff
@@ -48,5 +48,28 @@
...
+
+  if valid_system_commands.count > 0
+    valid_system_commands.each do |val_sys_cmd|
+      describe file(val_sys_cmd) do
+        it { should_not be_more_permissive_than('0755') }
+      end
+    end
+  else
+    describe "No Files Found" do
+      skip "No system commands found in /bin, /sbin, /usr/bin, /usr/sbin, /usr/local/bin or /usr/local/sbin, that are less permissive than 0755"
```
Comment (aaronlippold) [inline controls/V-75611.rb]: This should pass if No Files Found? or is it NA?

File: controls/V-75613.rb [hunk 10]
```diff
@@ -47,5 +47,28 @@
...
+
+  if valid_system_commands.count > 0
+    valid_system_commands.each do |val_sys_cmd|
+      describe file(val_sys_cmd) do
+        its('owner') { should cmp 'root' }
+      end
+    end
+  else
+    describe "No Files Found" do
+      skip "No system commands found in /bin, /sbin, /usr/bin, /usr/sbin, /usr/local/bin or /usr/local/sbin, that are NOT owned by root"
```
Comment (aaronlippold) [inline controls/V-75613.rb]: Same as above

File: controls/V-75615.rb [hunk 11]
```diff
@@ -48,5 +48,28 @@
...
+
+  if valid_system_commands.count > 0
+    valid_system_commands.each do |val_sys_cmd|
+      describe file(val_sys_cmd) do
+        its('group') { should cmp 'root' }
+      end
+    end
+  else
+    describe "No Files Found" do
+      skip "No system commands found in /bin, /sbin, /usr/bin, /usr/sbin, /usr/local/bin or /usr/local/sbin, that are NOT group-owned by root"
```
Comment (aaronlippold) [inline controls/V-75615.rb]: Same as above

File: controls/V-75811.rb [hunk 12]
```diff
@@ -57,5 +57,19 @@
...
+  if (lines.count > 0)
+    lines.each do |line|
+      dir = line.strip
+      describe directory(dir) do
+        it { should be_sticky }
+      end
+    end
+  else
+    describe "Sticky bit has been set on all world writable directories" do
+      skip ('Skipping tests as a sticky bit has been set on all world writable directories')
```
Comment (aaronlippold) [inline controls/V-75811.rb]: Same as other, does this pass or fail or is it NA if the sticky bit is set

File: controls/V-75813.rb [hunk 13]
```diff
@@ -73,5 +73,19 @@
...
+  if is_system_networked
+    describe ntp_conf do
+      it { should exist }
+      its('maxpoll') { should cmp 17 }
+      its('server') { should_not be_empty }
+      its('server') { should_not eq nil }
+    end
+  else
+    describe "System is not networked" do
+      skip "NTP checks skipped as the system is not networked"
```
Comment (aaronlippold) [inline controls/V-75813.rb]: I would assume this is NA if the system is not networked correct?

File: controls/V-75873.rb [hunk 14]
```diff
@@ -44,5 +44,9 @@
 line in \"/etc/sysctl.conf\" or in the appropriate file under \"/etc/sysctl.d\":
 
 net.ipv4.conf.all.accept_source_route=0"
+
+  describe command('sysctl net.ipv4.conf.all.accept_source_route') do
```
Comment (aaronlippold) [inline controls/V-75873.rb]: These type should be using the `kernel_paramater` resource correct?

File: controls/V-75875.rb [hunk 15]
```diff
@@ -44,5 +44,9 @@
 line in \"/etc/sysctl.conf\" or in the appropriate file under \"/etc/sysctl.d\":
 
 net.ipv4.conf.default.accept_source_route=0"
+
+  describe command('sysctl net.ipv4.conf.default.accept_source_route') do
```
Comment (aaronlippold) [inline controls/V-75875.rb]: These type should be using the kernel_paramater resource correct?

File: controls/V-75877.rb [hunk 16]
```diff
@@ -44,5 +44,9 @@
 line in \"/etc/sysctl.conf\" or in the appropriate file under \"/etc/sysctl.d\":
 
 net.ipv4.icmp_echo_ignore_broadcasts=1"
+
+  describe command('sysctl net.ipv4.icmp_echo_ignore_broadcasts') do
```
Comment (aaronlippold) [inline controls/V-75877.rb]: These type should be using the kernel_paramater resource correct? Double check please.

File: controls/V-75879.rb [hunk 17]
```diff
@@ -47,5 +47,9 @@
 line in \"/etc/sysctl.conf\" or in the appropriate file under \"/etc/sysctl.d\":
 
 net.ipv4.conf.default.accept_redirects=0"
+
+  describe command('sysctl net.ipv4.conf.default.accept_redirects') do
```
Comment (aaronlippold) [inline controls/V-75879.rb]: These type should be using the kernel_paramater resource correct? Double check please.

File: controls/V-75883.rb [hunk 18]
```diff
@@ -47,5 +47,9 @@
 line in \"/etc/sysctl.conf\" or in the appropriate file under \"/etc/sysctl.d\":
 
 net.ipv4.conf.default.send_redirects=0"
+
+  describe command('sysctl net.ipv4.conf.default.send_redirects') do
```
Comment (aaronlippold) [inline controls/V-75883.rb]: These type should be using the kernel_paramater resource correct? Double check please.

File: controls/V-75887.rb [hunk 19]
```diff
@@ -43,5 +43,9 @@
 line in \"/etc/sysctl.conf\" or in the appropriate file under \"/etc/sysctl.d\":
 
 net.ipv4.ip_forward=0"
+
+  describe command('sysctl net.ipv4.ip_forward') do
```
Comment (aaronlippold) [inline controls/V-75887.rb]: These type should be using the kernel_paramater resource correct? Double check please.

File: controls/V-75891.rb [hunk 20]
```diff
@@ -49,5 +49,20 @@
...
+  is_postfix_installed = package('postfix').installed?
+
+  if is_postfix_installed
+    postconf_output = command('postconf -n smtpd_client_restrictions').stdout.strip
+    smtpd_relay_restrictions = postconf_output.split(' = ')[1].split(', ')
+    describe smtpd_relay_restrictions do
+      it { should be_in ["permit_mynetworks", "permit_sasl_authenticated", "reject"] }
+    end
+  else
+    describe "Control Not Applicable as postfix is not installed" do
```
Comment (aaronlippold) [inline controls/V-75891.rb:62]: you need to update the impact for all cases where a control is Not Applicable to `impact 0`

File: controls/V-75893.rb [hunk 21]
```diff
@@ -51,5 +51,18 @@
...
 postmaster: root"
+
+  is_postfix_installed = package('postfix').installed?
+
+  if is_postfix_installed
+    describe command('grep "postmaster: *root$" /etc/aliases') do
+      its('stdout') { should_not be_empty }
+    end
+  else
+    describe "Control Not Applicable as postfix is not installed" do
```
Comment (aaronlippold) [inline controls/V-75893.rb:62]: you need to update the impact for all cases where a control is Not Applicable to impact 0

File: controls/V-75605.rb [hunk 22]
```diff
@@ -51,5 +51,19 @@
 with a mode more permissive than 0755.
 
 # sudo chmod 0755 [file]"
+
+  library_files = command('find /lib /lib64 /usr/lib -perm /022 -type f').stdout.strip.split("\n").entries
+
+  if library_files.count > 0
+    library_files.each do |lib_file|
+      describe file(lib_file) do
+        its('group') { should cmp 'root' }
```
Comment (aaronlippold) [inline controls/V-75605.rb]: @djhaynes does this make sense?

**Review by aaronlippold — CHANGES_REQUESTED** [review]

File: controls/V-75605.rb (hunk 1 above)
Comment (kkola) [reply controls/V-75605.rb]: describe block modified to account for passing the control

File: controls/V-75609.rb (hunk 3 above)
Comment (kkola) [reply controls/V-75609.rb]: describe block modified to account for passing the control

File: controls/V-75607.rb (hunk 2 above)
Comment (kkola) [reply controls/V-75607.rb]: describe block modified to account for passing the control

File: controls/V-75575.rb (hunk 4 above)
Comment (kkola) [reply controls/V-75575.rb]: describe block modified to account for passing the control

File: controls/V-75579.rb (hunk 5 above)
Comment (kkola) [reply controls/V-75579.rb]: describe block modified to account for passing the control

File: controls/V-75581.rb (hunk 6 above)
Comment (kkola) [reply controls/V-75581.rb]: describe block modified to account for passing the control

File: controls/V-75583.rb (hunk 7 above)
Comment (kkola) [reply controls/V-75583.rb]: describe block modified to account for passing the control

File: controls/V-75585.rb (hunk 8 above)
Comment (kkola) [reply controls/V-75585.rb]: describe blocks modified to account for kdump service requirement

File: controls/V-75611.rb (hunk 9 above)
Comment (kkola) [reply controls/V-75611.rb]: describe block modified to account for passing the control

File: controls/V-75615.rb (hunk 11 above)
Comment (kkola) [reply controls/V-75615.rb]: describe block modified to account for passing the control

File: controls/V-75813.rb (hunk 13 above)
Comment (kkola) [reply controls/V-75813.rb]: describe block modified to account for passing the control

File: controls/V-75873.rb (hunk 14 above)
Comment (kkola) [reply controls/V-75873.rb]: describe block modified to use kernel_parameter

File: controls/V-75877.rb (hunk 16 above)
Comment (kkola) [reply controls/V-75877.rb]: describe block modified to use kernel_parameter

File: controls/V-75879.rb (hunk 17 above)
Comment (kkola) [reply controls/V-75879.rb]: describe block modified to use kernel_parameter

File: controls/V-75883.rb (hunk 18 above)
Comment (kkola) [reply controls/V-75883.rb]: describe block modified to use kernel_parameter

File: controls/V-75887.rb (hunk 19 above)
Comment (kkola) [reply controls/V-75887.rb]: describe block modified to use kernel_parameter

File: controls/V-75613.rb (hunk 10 above)
Comment (kkola) [reply controls/V-75613.rb]: describe block modified to account for passing the control

File: controls/V-75811.rb (hunk 12 above)
Comment (kkola) [reply controls/V-75811.rb]: describe block modified to account for passing the control

File: controls/V-75875.rb (hunk 15 above)
Comment (kkola) [reply controls/V-75875.rb]: describe block modified to use kernel_parameter

File: controls/V-75891.rb (hunk 20 above)
Comment (kkola) [reply controls/V-75891.rb:62]: @aaronlippold is this just a matter of changing the impact value from 0.5 to 0 within the control or is there a programmatic method of taking care of this in the describe block?

File: controls/V-75893.rb (hunk 21 above)
Comment (kkola) [reply controls/V-75893.rb:62]: @aaronlippold is this just a matter of changing the impact value from 0.5 to 0 within the control or is there a programmatic method of taking care of this in the describe block?

**Pull request merged**